src/Stratis.Bitcoin/Consensus/ConsensusManagerBehavior.cs

```diff
@@ -386,10 +386,10 @@ private async Task<ConnectNewHeadersResult> PresentHeadersLockedAsync(List<Block
                 // Resync in case can't connect.
                 await this.ResyncAsync().ConfigureAwait(false);
             }
-            catch (ConsensusErrorException exception)
+            catch (HeaderInvalidException)
```

Comment:
I find the mismatch between ConsensusErrors.BannedHash.Throw and HeaderInvalidException awkward to follow. I assume the .Throw throws that exception but I would say that it isn’t obvious.

Reply:
They sort of have the same meaning so `HeaderInvalidException` is more general.